src/main/java/seedu/pill/util/Parser.java

```diff
@@ -117,6 +150,202 @@ public void parseCommand(String input) {
+     *                           argument or an invalid order type is provided.</li>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.INVALID_ITEM_FORMAT} if item
+     *                           details are incorrectly formatted
+     *                               (such as an invalid quantity or date format).</li>
+     *                       </ul>
+     *
+     *                       <p>During item processing, the method:
+     *                       <ul>
+     *                           <li>Parses each item's name, quantity, and optional expiry date from user input.</li>
+     *                           <li>Assumes a default quantity of 1 if no quantity is specified.</li>
+     *                           <li>Checks for proper date and quantity formats and throws an exception if they are
+     *                           invalid.</li>
+     *                       </ul>
+     *                       </p>
+     */
+    private OrderCommand parseOrderCommand(String arguments) throws PillException {
+        String[] commandArguments = arguments.split("\\s+");
+        if (commandArguments.length > 2) {
+            throw new PillException(ExceptionMessages.TOO_MANY_ARGUMENTS);
+        } else if (commandArguments.length < 2) {
+            throw new PillException(ExceptionMessages.INVALID_ORDER_COMMAND);
+        }
+        ItemMap itemsToOrder = new ItemMap();
+        OrderType orderType = null;
+        int numberOfItems = parseQuantity(commandArguments[1]);
+        if (commandArguments[0].equalsIgnoreCase("PURCHASE")) {
+            orderType = OrderType.PURCHASE;
+        } else if (commandArguments[0].equalsIgnoreCase("DISPENSE")) {
+            orderType = OrderType.DISPENSE;
+        } else {
+            throw new PillException(ExceptionMessages.INVALID_ORDER_COMMAND);
+        }
+        for (int i = 0; i < numberOfItems; i++) {
+            String userInput = ui.getInput();
+            String[] itemArguments = userInput.split("\\s+");
+            if (itemArguments.length < 2) {
+                throw new PillException(ExceptionMessages.INVALID_ITEM_FORMAT);
+            }
+
+            Integer quantityIndex = null;
+            Integer dateIndex = null;
+
+            for (int j = 0; j < itemArguments.length; j++) {
+                String currentArgument = itemArguments[j];
+
+                if (isValidDate(currentArgument)) {
+                    if (dateIndex != null) {
+                        throw new PillException(ExceptionMessages.INVALID_ITEM_FORMAT);
+                    }
+                    dateIndex = j;
+
+                    if (j != itemArguments.length - 1) {
+                        throw new PillException(ExceptionMessages.INVALID_ITEM_FORMAT);
+                    }
+                }
+
+                if (isANumber(currentArgument)) {
+                    quantityIndex = j;
+                }
+            }
```

Comment:
Perhaps there can be a way to abstract this part out to a separate helper function?

Reply:
Extracted out in 68759bf